core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1699,4 +1710,89 @@ public function generateRecordLabel(array $row, string $table = null, bool $prot
 
 		return $label;
 	}
+
+	/**
+	 * @param array<int|string> $ids
+	 */
+	protected function preloadCurrentRecords(array $ids, string $table): void
+	{
+		// Clear current cache to make sure records are gone if they cannot be loaded from the database below
+		foreach ($ids as $id)
+		{
+			$this->setCurrentRecordCache($id, $table, null);
+		}
+
+		/** @var Connection $connection */
+		$connection = System::getContainer()->get('database_connection');
+
+		$stmt = $connection->executeQuery(
+			'SELECT * FROM ' . $table . ' WHERE id IN (?)',
+			array($ids),
+			array(is_numeric(array_shift($ids)) ? Connection::PARAM_INT_ARRAY : Connection::PARAM_STR_ARRAY)
```

Comment:
Modifying `$ids` here seems confusing to me (and might have the potential for future bugs if `$ids` gets used below this line).
```suggestion
			array(is_numeric(array_values($ids)[0]) ? Connection::PARAM_INT_ARRAY : Connection::PARAM_STR_ARRAY)
```